common/src/main/java/com/sequenceiq/cloudbreak/quartz/metric/JobCountMetricConfigurator.java

```diff
@@ -24,20 +24,15 @@ public class JobCountMetricConfigurator {
     private GroupNameToJobCountFunction groupNameToJobCountFunction;
 
     @Inject
-    private Scheduler scheduler;
+    private List<MetricService> metricServices;
 
     @Inject
-    private MetricService metricService;
+    private List<? extends JobSchedulerService> jobSchedulerServices;
 
     @PostConstruct
     public void init() {
-        try {
-            for (String groupName : scheduler.getJobGroupNames()) {
-                metricService.registerGaugeMetric(QuartzMetricType.JOB_COUNT, groupName, groupNameToJobCountFunction,
-                        Map.of(JOB_GROUP.name(), groupName));
-            }
-        } catch (SchedulerException e) {
-            LOGGER.error("Cannot create job count metrics", e);
-        }
-    }
+        jobSchedulerServices.forEach(jobSchedulerService ->
+                metricServices.forEach(metricService -> metricService.registerGaugeMetric(QuartzMetricType.JOB_COUNT, jobSchedulerService.getJobGroup(),
+                        groupNameToJobCountFunction, Map.of(JOB_GROUP.name(), jobSchedulerService.getJobGroup()))));
```

Comment:
catch isn't needed anymore? in case of a failure further registrations will be skipped, won't they?

Reply:
Catch was for the scheduler call only, I have removed it thus its not needed anymore.

Reply:
ok,then this should be fine,thanks for the clarification